src/Sylius/Bundle/ApiBundle/spec/ApiPlatform/ConfigMergeManagerSpec.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace spec\Sylius\Bundle\ApiBundle\ApiPlatform;
+
+use PhpSpec\ObjectBehavior;
+
+class ConfigMergeManagerSpec extends ObjectBehavior
+{
+    public function it_merges_configs(): void
+    {
+        $this->mergeConfigs(
+            [
+                'admin_get' => [
+                    'method' => 'GET',
+                    'path' => 'admin/path/{tokenValue}'
+                ],
+                'admin_post' => [
+                    'method' => 'POST',
+                    'path' => 'admin/path'
+                ]
+            ],
+            [
+                'admin_patch' => [
+                    'method' => 'PATCH',
+                    'path' => 'admin/path/patch'
+                ],
+            ]
+        )->shouldReturn(
+            [
+                'admin_get' => [
+                    'method' => 'GET',
+                    'path' => 'admin/path/{tokenValue}'
+                ],
+                'admin_post' => [
+                    'method' => 'POST',
+                    'path' => 'admin/path'
+                ],
+                'admin_patch' => [
+                    'method' => 'PATCH',
+                    'path' => 'admin/path/patch'
+                ],
+            ]
+        );
+    }
+
+    public function it_removes_config_with_unset_keyword(): void
```

Comment:
We are missing case with `unset` and declared same endpoint again. Or at least the fragment when the following code will be executed:
```
if (null === $newValue) {
    continue;
}
```

I'm also pretty sure we missed most of the cases from this fragment:
![image](https://user-images.githubusercontent.com/6213903/122283513-e97fc280-ceec-11eb-9800-92dc9808b9b0.png)

Reply:
if (null === $newValue) {
    continue;
} 
is executed every time we declare endpoint with (unset) keyword with null value

Reply:
added few more cases to cover all of these